src/cascadia/LocalTests_SettingsModel/SerializationTests.cpp

```diff
@@ -0,0 +1,279 @@
+            "}"
+        };
+
+        const std::string smallGlobalsString{
+            "{"
+            "\"defaultProfile\": \"{61c54bbd-c2c6-5271-96e7-009a87ff44bf}\""
+            "}"
+        };
+
+        auto roundtripTest = [this](auto jsonString) {
+            const auto json{ VerifyParseSucceeded(jsonString) };
+            const auto globals{ implementation::GlobalAppSettings::FromJson(json) };
+
+            const auto result{ globals->ToJson() };
+
+            VERIFY_ARE_EQUAL(jsonString, toString(result));
+        };
+
+        roundtripTest(globalsString);
+        roundtripTest(smallGlobalsString);
+    }
+
+    void SerializationTests::Profile()
+    {
+        // This needs to be in alphabetical order.
+        const std::string profileString{
+            "{"
+            "\"altGrAliasing\": true,"
+            "\"antialiasingMode\": \"grayscale\","
+            "\"background\": \"#BBCCAA\","
+            "\"backgroundImage\": \"made_you_look.jpeg\","
+            "\"backgroundImageAlignment\": \"center\","
+            "\"backgroundImageOpacity\": 1.0,"
+            "\"backgroundImageStretchMode\": \"uniformToFill\","
+            "\"closeOnExit\": \"graceful\","
+            "\"colorScheme\": \"Campbell\","
+            "\"commandline\": \"%SystemRoot%\\\\System32\\\\WindowsPowerShell\\\\v1.0\\\\powershell.exe\","
+            "\"cursorColor\": \"#CCBBAA\","
+            "\"cursorHeight\": 10,"
+            "\"cursorShape\": \"bar\","
+            "\"experimental.retroTerminalEffect\": false,"
+            "\"fontFace\": \"Cascadia Mono\","
+            "\"fontSize\": 12,"
+            "\"fontWeight\": \"normal\","
+            "\"foreground\": \"#AABBCC\","
+            "\"guid\": \"{61c54bbd-c2c6-5271-96e7-009a87ff44bf}\","
+            "\"hidden\": false,"
+            "\"historySize\": 9001,"
+            "\"icon\": \"ms-appx:///ProfileIcons/{61c54bbd-c2c6-5271-96e7-009a87ff44bf}.png\","
+            "\"name\": \"Windows PowerShell\","
+            "\"padding\": \"8, 8, 8, 8\","
+            "\"scrollbarState\": \"visible\","
+            "\"selectionBackground\": \"#CCAABB\","
+            "\"snapOnInput\": true,"
+            "\"startingDirectory\": \"%USERPROFILE%\","
+            "\"suppressApplicationTitle\": false,"
+            "\"tabColor\": \"#0C0C0C\","
+            "\"tabTitle\": \"Cool Tab\","
+            "\"useAcrylic\": false"
+            "}"
+        };
+
+        const std::string smallProfileString{
+            "{"
+            "\"name\": \"Custom Profile\""
+            "}"
+        };
+
+        auto roundtripTest = [this](auto jsonString) {
+            const auto json{ VerifyParseSucceeded(jsonString) };
+            const auto profile{ implementation::Profile::FromJson(json) };
+
+            const auto result{ profile->ToJson() };
+
+            VERIFY_ARE_EQUAL(jsonString, toString(result));
+        };
+
+        roundtripTest(profileString);
+        roundtripTest(smallProfileString);

```

Comment:
Can we get tests for roundtripping `foreground` (which is weird), `tabColor` (also weird) and maybe also `source`?